src/coprocessor/codec/chunk/column.rs

```diff
@@ -0,0 +1,425 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Result;
+use coprocessor::codec::Datum;
+use coprocessor::codec::mysql::decimal::DECIMAL_STRUCT_SIZE;
+use coprocessor::codec::mysql::{types, Decimal, DecimalEncoder, Duration, DurationEncoder, Json,
+                                JsonDecoder, JsonEncoder, Time, TimeEncoder};
+use tipb::expression::FieldType;
+use util::codec::number::{NumberDecoder, NumberEncoder};
+#[derive(Default)]
+pub struct Column {
+    length: usize,
+    null_cnt: usize,
+    null_bitmap: Vec<u8>,
+    var_offsets: Vec<usize>,
+    data: Vec<u8>,
+    // if the data's length is fixed, fixed_len should be bigger than 0
+    fixed_len: usize,
```

Comment:
why is this field different from tikv's?

Reply:
Since `fixed_len` is more clear than `elemBuf    []byte` here, also it will be much more complex if implement as tidb.